build_runner/bin/graph_inspector.dart

```diff
@@ -0,0 +1,203 @@
+    }
+    for (var stringUri in stringUris) {
+      var id = _getIdFromStringUri(stringUri);
+      if (id == null) {
+        continue;
+      }
+      var node = assetGraph.get(id);
+      if (node == null) {
+        stderr.writeln('Unable to find an asset node for $stringUri.');
+        continue;
+      }
+
+      var description = new StringBuffer()
+        ..writeln('Asset: $stringUri')
+        ..writeln('  type: ${node.runtimeType}');
+
+      if (node is GeneratedAssetNode) {
+        description.writeln('  wasOutput: ${node.wasOutput}');
+      }
+
+      _printAsset(AssetId asset) =>
+          _listAsset(asset, description, indentation: '    ');
+
+      if (argResults['verbose'] == true) {
+        description.writeln('  primary outputs:');
+        node.primaryOutputs.forEach(_printAsset);
+
+        description.writeln('  secondary outputs:');
+        node.outputs.difference(node.primaryOutputs).forEach(_printAsset);
+
+        if (node is GeneratedAssetNode) {
+          description.writeln('  inputs:');
+          var inputs = assetGraph.allNodes
+              .where((n) => n.outputs.contains(node.id))
+              .map((n) => n.id);
+          inputs.forEach(_printAsset);
+        }
+      }
+
+      stdout.write(description);
+    }
+  }
+}
+
+class GraphCommand extends Command {
+  @override
+  String get name => 'graph';
+
+  @override
+  String get description => 'Lists all the nodes in the graph.';
+
+  @override
+  String get invocation => '${super.invocation} <dart-uri>';
+
+  GraphCommand() {
+    argParser.addFlag('generated',
+        abbr: 'g', help: 'Show only generated assets.', defaultsTo: false);
+    argParser.addFlag('original',
+        abbr: 'o',
+        help: 'Show only original source assets.',
+        defaultsTo: false);
+    argParser.addOption('package',
+        abbr: 'p', help: 'Filters nodes to a certain package');
+    argParser.addOption('pattern',
+        abbr: 'm', help: 'glob pattern for path matching');
+  }
+
+  @override
+  run() {
+    var showGenerated = argResults['generated'] as bool;
+    var showSources = argResults['original'] as bool;
+    Iterable<AssetNode> nodes;
+    if (showGenerated) {
+      nodes = assetGraph.allNodes.where((n) => n is GeneratedAssetNode);
+    } else if (showSources) {
+      nodes = assetGraph.allNodes.where((n) => n is! GeneratedAssetNode);
+    } else {
+      nodes = assetGraph.allNodes;
+    }
+
+    var package = argResults['package'] as String;
+    if (package != null) {
+      nodes = nodes.where((node) => node.id.package == package);
+    }
+
+    var pattern = argResults['pattern'] as String;
+    if (pattern != null) {
+      var glob = new Glob(pattern);
+      nodes = nodes.where((node) => glob.matches(node.id.path));
+    }
+
+    for (var node in nodes) {
+      _listAsset(node.id, stdout, indentation: '  ');
+    }
+  }
+}
+
+AssetId _getIdFromStringUri(String stringUri) {
```

Comment:
[nit] would prefer without `get`. `_idFromString` is clear to me.

Reply:
done